File: src/main/java/codes/qdbp/serverplugin/recipes/LightRecipe.java
package codes.qdbp.serverplugin.recipes;

import org.bukkit.Material;
import org.bukkit.NamespacedKey;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.ShapedRecipe;

public class LightRecipe {

    ShapedRecipe lightRecipe;

    public LightRecipe(NamespacedKey namespacedKey) {
        ItemStack light = new ItemStack(Material.LIGHT, 1);

        this.lightRecipe = new ShapedRecipe(namespacedKey, light);

        this.lightRecipe.shape(
                "ttt",
                "tit",
                "ttt"
        );
        this.lightRecipe.setIngredient('t', Material.TORCH);
        this.lightRecipe.setIngredient('i', Material.IRON_BLOCK);
    }

    public ShapedRecipe getLightRecipe() {
        return lightRecipe;
    }
}

File: src/main/java/codes/qdbp/serverplugin/recipes/LightRecipe.kt
package codes.qdbp.serverplugin.recipes

import codes.qdbp.serverplugin.Serverplugin
import org.bukkit.Material
import org.bukkit.NamespacedKey
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.ShapedRecipe


data class LightRecipe(val plugin: Serverplugin){

    val lightRecipe =
        ShapedRecipe(NamespacedKey(plugin, "light"), ItemStack(Material.LIGHT))
            .shape(
                "ttt",
                "tit",
                "ttt"
            )
            .setIngredient('t', Material.TORCH)
            .setIngredient('i', Material.IRON_BLOCK)

}